src/moscot/problems/time/_mixins.py

```diff
@@ -427,20 +427,19 @@ def pull(
     @property
     def prior_growth_rates(self: TemporalMixinProtocol[K, B]) -> Optional[pd.DataFrame]:
         """Return the prior estimate of growth rates of the cells in the source distribution."""
-        # TODO(michalk8): FIXME
         cols = ["prior_growth_rates"]
         df_list = [
             pd.DataFrame(problem.prior_growth_rates, index=problem.adata.obs.index, columns=cols)
             for problem in self.problems.values()
         ]
-        tup = list(self)[-1]
+        indices_remaining = set(self.adata.obs_names) - {ind for df in df_list for ind in df.index}
         df_list.append(
             pd.DataFrame(
                 np.full(
-                    shape=(len(self.problems[tup].adata_tgt.obs), 1),
+                    shape=(len(indices_remaining), 1),
```

Comment:
For readability, `shape=len(indices_remaining), len(columns)`.
But I think `pandas` should be able to broadcast this:
```python
pd.DataFrame(np.nan, index=list(indices_remaining), columns=columns)
```